frontend/src/lib/k8s/cluster.ts

```diff
@@ -97,6 +97,17 @@ export function makeKubeObject<T extends KubeObjectInterface | KubeEvent>(
       return this._class().className;
     }
 
+    static get pluralName(): string {
+      // This is a naive way to get the plural name of the object by default. It will
+      // work in most cases, but for exceptions (like Ingress), we must override this.
+      return this.className.toLowerCase() + 's';
+    }
+
+    get pluralName(): string {
+      // In case we need to override the plural name in instances.
+      return this._class().pluralName;
+    }
+
     get listRoute(): string {
       return this.detailsRoute + 's';
```

Comment:
Should the pluralName be used here too?

Reply:
good point. Maybe it should. I will update.